Docker tests failing with JSON errors on openSuSE

## Version of Singularity:

singularity version 3.0.0-alpha.1-217-g41c253f6-dirty

### Expected behavior

Docker tests pass.

### Actual behavior

All the tests that use `singularity --build docker://` seem to fail on openSuSE. They pass without issue on CentOS.

=== RUN   TestDockerPermissions
--- FAIL: TestDockerPermissions (0.01s)
    docker_test.go:96: ^[[33mWARNING:^[[0m Authentication token file not found : Only pulls of public images will succeed
        2018/09/12 06:18:38 Error initializing source docker://dctrud/docker-singularity-userperms:latest: error getting username and password: error reading JSON file "/root/.docker/config.json": error unmarshaling JSON at "/root/.docker/config.json": unexpected end of JSON input

docker_test.go:97: unexpected failure: exit status 1

---

=== RUN   TestDockerWhiteoutSymlink
--- FAIL: TestDockerWhiteoutSymlink (0.01s)
    docker_test.go:148: ^[[33mWARNING:^[[0m Authentication token file not found : Only pulls of public images will succeed
        2018/09/12 06:39:44 Error initializing source docker://dctrud/docker-singularity-linkwh:latest: error getting username and password: error reading JSON file "/root/.docker/config.json": error unmarshaling JSON at "/root/.docker/config.json": unexpected end of JSON input

docker_test.go:149: unexpected failure: exit status 1

---

=== RUN   TestSandboxAssemblerDocker
^[[0mDEBUG   ^[[0m[U=1000,P=20289]   NewBundle()                   Created temporary directory for bundle /tmp/sbuild-oci-011090937
2018/09/12 06:44:44 Error initializing source docker://alpine:latest: error getting username and password: error reading JSON file "/root/.docker/config.json": error unmarshaling JSON at "/root/.docker/config.json": unexpected end of JSON input
FAIL    github.com/singularityware/singularity/src/pkg/build/assemblers 0.038s

---

=== RUN   TestOCIConveyorDocker
^[[0mDEBUG   ^[[0m[U=1000,P=20318]   NewBundle()                   Created temporary directory for bundle /tmp/sbuild-oci-390489775
2018/09/12 06:44:48 Error initializing source docker://alpine:latest: error getting username and password: error reading JSON file "/root/.docker/config.json": error unmarshaling JSON at "/root/.docker/config.json": unexpected end of JSON input
FAIL    github.com/singularityware/singularity/src/pkg/build/sources    1.859s

---

### Steps to reproduce behavior

Run `make test` after building on openSuSE. Probably has the same issues on SLE, but did not verify.

--- a/src/pkg/test/privilege.go
+++ b/src/pkg/test/privilege.go
@@ -12,6 +12,7 @@ import (
 	"os"
 	"os/user"
 	"runtime"
+	"strconv"
 	"syscall"
 	"testing"
 )
@@ -133,7 +134,7 @@ func getUnprivIDs(pid int) (uid int, gid int) {
 func init() {
 	origUID = os.Getuid()
 	origGID = os.Getgid()
-	origUser, err := user.LookupId(fmt.Sprintf("%d", origUID))
+	origUser, err := user.LookupId(strconv.Itoa(origUID))
 
 	if err != nil {
 		log.Fatalf("err: %s", err)
@@ -142,7 +143,7 @@ func init() {
 	origHome = origUser.HomeDir
 
 	unprivUID, unprivGID = getUnprivIDs(os.Getpid())
-	unprivUser, err := user.LookupId(fmt.Sprintf("%d", unprivUID))
+	unprivUser, err := user.LookupId(strconv.Itoa(unprivUID))
 
 	if err != nil {
 		log.Fatalf("err: %s", err)

--- a/src/pkg/test/privilege.go
+++ b/src/pkg/test/privilege.go
@@ -10,12 +10,14 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"os/user"
 	"runtime"
 	"syscall"
 	"testing"
 )
 
 var origUID, origGID, unprivUID, unprivGID int
+var origHome, unprivHome string
 
 // EnsurePrivilege ensures elevated privileges are available during a test.
 func EnsurePrivilege(t *testing.T) {
@@ -43,6 +45,10 @@ func DropPrivilege(t *testing.T) {
 		if err := syscall.Setresuid(unprivUID, unprivUID, origUID); err != nil {
 			t.Fatalf("failed to set user identity: %v", err)
 		}
+
+		if err := os.Setenv("HOME", unprivHome); err != nil {
+			t.Fatalf("failed to set HOME environment variable: %v", err)
+		}
 	}
 }
 
@@ -54,6 +60,10 @@ func ResetPrivilege(t *testing.T) {
 	if err := syscall.Setresgid(origGID, origGID, unprivGID); err != nil {
 		t.Fatalf("failed to reset group identity: %v", err)
 	}
+	if err := os.Setenv("HOME", origHome); err != nil {
+		t.Fatalf("failed to reset HOME environment variable: %v", err)
+	}
+
 	runtime.UnlockOSThread()
 }
 
@@ -123,5 +133,20 @@ func getUnprivIDs(pid int) (uid int, gid int) {
 func init() {
 	origUID = os.Getuid()
 	origGID = os.Getgid()
+	origUser, err := user.LookupId(fmt.Sprintf("%d", origUID))
+
+	if err != nil {
+		log.Fatalf("err: %s", err)
+	}
+
+	origHome = origUser.HomeDir
+
 	unprivUID, unprivGID = getUnprivIDs(os.Getpid())
+	unprivUser, err := user.LookupId(fmt.Sprintf("%d", unprivUID))
+
+	if err != nil {
+		log.Fatalf("err: %s", err)
+	}
+
+	unprivHome = unprivUser.HomeDir
 }

